src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -306,14 +329,11 @@ private void validateObjectKey(String key, JsonNode value, int depth, int parent
         throw ErrorCode.SHRED_DOC_KEY_NAME_VIOLATION.toApiException("empty names not allowed");
       }
       if (!DocumentConstants.Fields.VALID_NAME_PATTERN.matcher(key).matches()) {
-        // Special names are accepted in some cases: for now only one such case for
-        // Date values -- if more needed, will refactor. But for now inline:
-        if (JsonUtil.EJSON_VALUE_KEY_DATE.equals(key) && value.isValueNode()) {
-          ; // Fine, looks like legit Date value
-        } else if (key.equals(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD) && depth == 1) {
-          ; // Fine, looks like legit vector property
-        } else if (key.equals(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD) && depth == 1) {
-          ; // Fine, looks like legit vectorize property
+        // Special names are accepted in some cases:
+        if ((depth == 1)
```

Comment:
`$date` value now validated in different place, along with new JSON Extension values.